### PR TITLE
WIP - Issue #54

### PR DESCRIPTION
**Pull request by derek-smith**
Issue #54 - Code out files and folders REST API

This is a work-in-progress pull request.

I've added TODO comments where properties/methods still need to be added. There ain't much logic in here yet. It's mostly just copy/paste work so far.

### CODE REVIEW

**Comment by derek-smith** [comment]
And I haven't created any unit tests yet either.

File: src/sharepoint/rest/folders.ts [hunk 1]
```diff
@@ -84,10 +149,44 @@ export class Folder extends QueryableInstance {
...
+                "X-HTTP-Method": "DELETE",
+            },
+        });
+    }
+
+    /**
+     * Moves the folder to the Recycle Bin and returns the identifier of the new Recycle Bin item.
+     */
+    public recycle(): Promise<string> {
+        this.append("recycle");
```
Comment (patrick-rodgers) [inline src/sharepoint/rest/folders.ts]: Want to get away from the this.append, I would create a new File(this, "recycle") and return the .post() of that. I know you got this from my code, but I updated it to maintain the "this" instance - so you can blame me :)

**Comment by patrick-rodgers** [comment]
Looks like a good start, I made a few comments. keep up the good work, excited to see this part finished up.

**Comment by derek-smith** [comment]
I think I've added all the properties and methods for files, folders, and versions.

I didn't include `getByUrl()` for `Files` or `Folders` because it [isn't recommended](https://msdn.microsoft.com/en-us/library/office/dn450841.aspx#bk_FileCollectionGetByUrl).

I'll be adding unit tests next, but that will not be until Monday or Tuesday (maybe tonight).

I've manually tested a lot of these APIs but not close to all of them. I'm sure there are some bugs in there. Any help with manual testing would be greatly appreciated!

99% of the jsdoc comments came from the MSDN reference.

**Comment by patrick-rodgers** [comment]
Great, thanks Derek - all my comments pretty much were just copied from there as well. Let me know when you are ready for a final review.

**Comment by derek-smith** [comment]
Ready for final review. Note that a lot of the properties & methods weren't tested against a SharePoint site. I don't know if we can merge it or not based on that?

**Comment by derek-smith** [comment]
I just didn't find the time to do all the testing...

**Comment by patrick-rodgers** [comment]
Thanks Derek, I looked over things and it looks good. I am going to merge it, but do please circle back and do some testing. I am finding small things here and there with my stuff as I work on samples and test - so always good if we can do deep testing. But also if there is a problem an issue can be created.

I'll close the issue as well, thanks for the work!

**Pull request merged**